backend/users-service/app/src/index.ts

```diff
@@ -1,7 +1,71 @@
-import { Elysia } from "elysia";
+import { Elysia, t } from 'elysia'
+import { swagger } from '@elysiajs/swagger'
+import { cors } from '@elysiajs/cors'
+import { helmet } from 'elysia-helmet'
+import jwt from '@elysiajs/jwt'
+import { PrismaClient } from '@prisma/client'
+import { logger } from '@bogeychan/elysia-logger'
+import { Role } from './constants'
 
-const app = new Elysia().get("/", () => "Users service").listen(80);
+const db = new PrismaClient()
+
+const app = new Elysia({ prefix: '/users' })
+  .use(cors())
+  .use(helmet())
+  .use(
+    jwt({
+      name: 'jwt',
+      secret: process.env.JWT_SECRET as string
+    })
+  )
+  .use(logger())
+  .use(swagger())
+  .get(
+    '/:id',
+    async ({ params: { id } }) =>
+      await db.user.findUnique({
+        where: {
+          id
+        },
+        include: {
+          questions: true
+        }
+      })
+  )
+  .post(
+    '/',
+    async ({ body: { avatarUrl, email } }) =>
+      await db.user.create({
+        data: {
+          avatarUrl,
+          email,
+          role: Role.User
+        }
+      }),
+    {
+      body: t.Object({ avatarUrl: t.String(), email: t.String() })
```

Comment:
btw where does `t` come from and what does it do? is this a request validation schema?

Reply:
okay i found that its from this library
https://www.npmjs.com/package/@sinclair/typebox#options

for the email, we can do further validation with
```
email: t.String({
  format: 'email'
})
```

dyt there's validation for URLs as well? i cant seem to find it in the docs

Reply:
> okay i found that its from this library https://www.npmjs.com/package/@sinclair/typebox#options
> 
> for the email, we can do further validation with
> 
> ```
> email: t.String({
>   format: 'email'
> })
> ```
> 
> dyt there's validation for URLs as well? i cant seem to find it in the docs

Nice catch will add that! Yeah it's not in the docs... will look into it

Reply:
Resolved by adding `email` and `avatarUrl` validation in 2c5145959e884a2c3d2e4cc9a7b9077359e7e45a